openstax/rex-web
# Issue 1519 - add update-archive-version script to handle page url changes after archive code version updates

for: https://github.com/openstax/unified/issues/1519

Testing:
- change value of `REACT_APP_ARCHIVE_URL` in `config.archive-url.json` to an empty string `""`
- remove all `.json` files from `/data/redirects/*`
- run `node ./script/entry.js update-archive-version --newArchiveUrl /apps/archive/20210224.204120`
- expected output should be similar to this:
```
Preparing books...
Updating redirects [===================                                         ] 19/60 (49.4s ETA) | Finished processing Principles of Accounting, Volume 1: Financial Accounting | 9ab4ba6d-1e48-486d-a2de-38ae16
Updating redirects [=========================                                   ] 25/60 (46.5s ETA) | Finished processing Principles of Accounting, Volume 2: Managerial Accounting | 920d1c8a-606c-4888-bfd4-d1ee2
Updating redirects [============================================================] 60/60 (0.0s ETA) | Finished processing Principles of Economics | 69619d2b-68f0-44b0-b074-a9b2bf90b2c6


Added 14 redirects for book Business Law I Essentials | 464a3fba-68c1-426a-99f9-597e739dc911
Added 11 redirects for book Business Ethics | 914ac66e-e1ec-486d-8a9c-97b0f7a99774
Added 1 redirects for book Introduction to Sociology 2e | 02040312-72c8-441e-a685-20e9333f3e1d
Added 1 redirects for book Chemistry: Atoms First | 4539ae23-1ccc-421e-9b25-843acbb6c4b0
Added 1 redirects for book Principles of Microeconomics | ea2f225e-6063-41ca-bcd8-36482e15ef65
Added 1 redirects for book Principles of Macroeconomics | 4061c832-098e-4b3c-a1d9-7eb593a2cb31
Updating config.archive-url.json with /apps/archive/20210224.204120
```
- run `git status`
- expected: no changes to commit
